client/src/components/DataFiles/DataFilesModals/DataFilesCompressModal.js

```diff
@@ -0,0 +1,197 @@
+                onChange={handleSelectChange}
+              >
+                <option value="zip">zip</option>
+                <option value="tar.gz">tar.gz</option>
+              </Input>
+            </FormGroup>
+          </Form>
+        );
+      }}
+    </Formik>
+  );
+};
+DataFilesCompressForm.propTypes = {
+  first: PropTypes.string.isRequired,
+  formRef: PropTypes.oneOfType([
+    PropTypes.func,
+    PropTypes.shape({ current: PropTypes.instanceOf(Element) })
+  ]).isRequired
+};
+
+const DataFilesCompressModal = () => {
+  const history = useHistory();
+  const location = useLocation();
+  const dispatch = useDispatch();
+  const status = useSelector(
+    state => state.files.operationStatus.compress,
+    shallowEqual
+  );
+
+  const params = useSelector(
+    state => state.files.params.FilesListing,
+    shallowEqual
+  );
+
+  const isOpen = useSelector(state => state.files.modals.compress);
+  const selectedFiles = useSelector(
+    ({ files: { selected, listing } }) =>
+      selected.FilesListing.map(i => ({
+        ...listing.FilesListing[i]
+      })),
+    shallowEqual
+  );
+  const selected = useMemo(() => selectedFiles, [isOpen]);
+  const formRef = React.useRef();
+  const toggle = () =>
+    dispatch({
+      type: 'DATA_FILES_TOGGLE_MODAL',
+      payload: { operation: 'compress', props: {} }
+    });
+
+  const onOpened = () => {
+    dispatch({
+      type: 'FETCH_FILES_MODAL',
+      payload: { ...params, section: 'modal' }
+    });
+  };
+
+  const onClosed = () => {
+    dispatch({ type: 'DATA_FILES_MODAL_CLOSE' });
+    if (isString(status)) {
+      dispatch({
+        type: 'DATA_FILES_SET_OPERATION_STATUS',
+        payload: { status: {}, operation: 'compress' }
+      });
+      history.push(location.pathname);
+    }
+  };
+
+  const compressCallback = () => {
+    const { filename } = formRef.current.values;
+    dispatch({
+      type: 'DATA_FILES_COMPRESS',
+      payload: { filename, files: selected }
+    });
+  };
+
+  let buttonIcon;
+  if (status === 'RUNNING') {
+    buttonIcon = <LoadingSpinner placement="inline" />;
+  } else if (status === 'ERROR') {
+    buttonIcon = <Icon name="alert" />;
+  } else {
+    buttonIcon = null;
+  }
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onOpened={onOpened}
+      onClosed={onClosed}
+      toggle={toggle}
+      className="dataFilesModal"
+    >
+      <ModalHeader toggle={toggle}>Compress Files</ModalHeader>
+      <ModalBody>
+        <DataFilesCompressForm
+          formRef={formRef}
+          first={(selectedFiles[0] && selectedFiles[0].name) || ''}
+        />
+        <p>
+          A job to compress your files will be submitted on your behalf. You can
+          check the status of this job on your Dashboard, and your compressed
+          file will appear in this directory.
+        </p>
+        {status === 'SUCCESS' && (
+          <span style={{ color: 'green' }}>
+            Successfully started compress job
```

Comment:
Is it okay for users to be able to run the compress job on the same files again?
![Screen Shot 2020-12-10 at 4 05 42 PM](https://user-images.githubusercontent.com/47395902/101835571-9a9e3e00-3b01-11eb-8c00-b93055bc0838.png)

Reply:
I wondering the same thing (same question for Extract Modal).  Would it be better to disable "Compress" button after successful submission.

Plus, should the message be to the left of the Compress button?

Reply:
We could probably do it with state.